File: Graph/1971. Find if Path Exists in Graph.py
class Solution:
    def validPath(self, n: int, edges: List[List[int]], start: int, end: int) -> bool:
        from queue import Queue
        visit = [0] * n
        g = [[] for _ in range(n)]
        for e in edges:
            g[e[0]].append(e[1])
            g[e[1]].append(e[0])
        
        q = Queue()
        visit[start] = 1
        q.put(start)
        
        while not q.empty():
            i = q.get()
            if i == end:
                return 1
            for n in g[i]:
                if visit[n] == 0:
                    visit[n] = 1
                    q.put(n)
        return 0

File: Graph/1971. Find if Path Exists in Graph.cpp
class Solution {
public:
    bool validPath(int n, vector<vector<int>>& edges, int start, int end) {
        vector<bool> visit(n, 0);
        vector<vector<int>> g(n);
        
        for (auto e : edges){
            g[e[0]].push_back(e[1]);
            g[e[1]].push_back(e[0]);
        }
        
        queue<int> q;
        q.push(start);
        visit[start] = 1;
        
        int i;
        while (not q.empty()){
            i = q.front();
            q.pop();
            if (i == end){
                return true;
            }
            for (auto n : g[i]){
                if (visit[n] == 0){
                    visit[n] = 1;
                    q.push(n);
                }
            }
        }
        return false;
        
    }
};